plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/local/LocalDockerModule.java

```diff
@@ -56,6 +57,8 @@ protected void configure() {
                   .to(org.eclipse.che.plugin.docker.machine.DefaultServerEvaluationStrategy.class);
         strategies.addBinding("docker-local")
                   .to(org.eclipse.che.plugin.docker.machine.LocalDockerServerEvaluationStrategy.class);
+        strategies.addBinding("single-port")
+                  .to(LocalDockerSinglePortServerEvaluationStrategy.class);
```

Comment:
This looks really similar to @benoitf 's CustomServerEvaluationStrategy what was the issue what it can't handle so you've created LocalDockerSinglePortServerEvaluationStrategy?